assertion-analyser.js

```diff
@@ -60,10 +62,10 @@ function objParser(str, init) {
 
 function replacer(str) {
   // replace objects with a symbol ( __#n)
-  var obj;
-  var cnt = 0;
-  var data = [];
-  while(obj = objParser(str)) {
+  let obj;
+  let cnt = 0;
+  let data = [];
+  while(obj == objParser(str)) {
```

Comment:
```suggestion
  while(obj = objParser(str)) {
```

Reply:
That looks like something that can be easily mistaken by someone down the lane as an incorrect conditional without context. Can this be made any simpler (even if it's not smart or is more code).

Reply:
Seeing as this is in many of the boilerplates, I would recommend we open an issue about this, first. So far, I just added a comment: https://github.com/freeCodeCamp/boilerplate-mochachai/pull/27/files#diff-c8580e1410209df94bd3f9675480f46dad5973282f1ef427f4cf85ef0a71580bR39

Reply:
@raisedadead This is only for maintainers.  The student does not change this file, so is it really necessary to change the working code?

Reply:
Why not:

```js
let obj = objParser(str);
while(obj) {
  data[cnt] = obj.obj;
  str = str.substring(0, obj.start) + '__#' + cnt++ + str.substring(obj.end+1);
  obj = objParser(str);
}
```

It's not the prettiest code, but it avoids the `=` vs `==` confusion.

Assuming we want to change things, I haven't been following this close enough to know.

Reply:
@ojeytonwilliams If you submit that as a suggestion I can commit it.  Have you tested locally to make sure it works?

Reply:
@SaintPeter and @ojeytonwilliams We are likely best not changing anything from what we know works. As Randell mentioned, it would be great to get an issue going to have these tester files be turned into a package, and avoid diverging all of these boilerplates.

Reply:
> This is only for maintainers. The student does not change this file, so is it really necessary to change the working code?

I do not have context of it, and that's the point. From a cursory look it looks like a missed / incorrect conditional statement. I understand it's just a smart way, but a few months down the lane it pretty easy t miss and someone can make the change for semantic reasons.

Often its best to make it foolproof (trading for more code).

Reply:
How many boilerplates use this code, because someone should create an issue to track all them where we need to do the same thing.

Reply:
I will get an issue put together 🙂

Reply:
If this is a common utility, we are happy to have this as a package. Let me know if you want to go that direction (not necessarily in this PR) and I will set things up. That said I would still recommend keeping it simple like Oliver mentioned in above. 

And more importantly I am totally fine not holding this PR for the nit-pick. It can totally be taken up in its own issue.

Reply:
Should I drop that commit or is it fine as is?

Reply:
I think we can leave it.  The change works and is documented in issue #40180.  I am going to try and review this PR today along with the corresponding `/learn` PR #39627 today or this weekend.